options.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package ringpop
+
+import (
+	"errors"
+	"io/ioutil"
+
+	"github.com/Sirupsen/logrus"
+	log "github.com/uber-common/bark"
+	"github.com/uber/ringpop-go/shared"
+)
+
+type Configuration struct {
+	// App is the name used to uniquely identify members of the same ring.
+	// Members will only talk to other members with the same app name. Note
+	// that App is taken as an argument of the Ringpop constructor and not a
+	// configuration option. This is to prevent accidental misconfiguration.
+	App string
+
+	ReplicaPoints int
+}
+
+// Options are modifiers that configure/modify a real Ringpop object
+type Option func(*Ringpop) error
+
+// applyOptions applies runtime configuration options to the specified Ringpop
+// instance.
+func applyOptions(r *Ringpop, opts []Option) error {
+	for _, option := range opts {
+		err := option(r)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// checkOptions checks that the Ringpop instance has been properly configured
+// with all the required options.
+func checkOptions(rp *Ringpop) []error {
+	errs := []error{}
```

Comment:
I feel like this function makes the assumption that our default options will always "do the right thing," meaning, if an identity resolver is required, we assume that there is a default that fills it in. Is this true? If so, I feel like this assumption is unsafe and that checkOptions should verify the existence of all required fields. Thoughts?